### PR TITLE
[WIP][SPARK-28894][SQL][TESTS] Report test results of SQLQueryTestSuite correctly in Jenkins

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

See https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109834/testReport/junit/org.apache.spark.sql/SQLQueryTestSuite/

Screen Shot 2019-08-28 at 4 08 58 PM

```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuite hostname="C02Y52ZLJGH5" name="org.apache.spark.sql.SQLQueryTestSuite" tests="3" errors="0" failures="0" skipped="0" time="14.475">
    ...
    <testcase classname="org.apache.spark.sql.SQLQueryTestSuite" name="sql - Scala UDF" time="6.703">
    </testcase>
    <testcase classname="org.apache.spark.sql.SQLQueryTestSuite" name="sql - Regular Python UDF" time="4.442">
    </testcase>
    <testcase classname="org.apache.spark.sql.SQLQueryTestSuite" name="sql - Scalar Pandas UDF" time="3.33">
    </testcase>
    <system-out/>
    <system-err/>
</testsuite>
```

Root cause seems sbt does not correctly generates the dot in its test name.

Also, seems Jenkins has multiple implicit rules for JUnit XML reports, in particular, about `classname` and `name` in `testcase` (e.g., the first dot is package name, etc.). (See for instance https://stackoverflow.com/questions/29471614/why-is-jenkins-improperly-parsing-my-junit-report).

This problem happens to all the tests written in Spark so we better have to avoid dot in test names; otherwise, the test title seems truncated in Jenkins test report.

### Why are the changes needed?

To debug Jenkins logs easier.

### Does this PR introduce any user-facing change?

It store the test name correctly in Jenkins.

### How was this patch tested?

Jenkins test https://github.com/apache/spark/pull/25608#issuecomment-525671194

Screen Shot 2019-08-28 at 8 10 40 PM

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
No haha. I wanted to verify if Jenkins shows the results properly or not. Let me add WIP in PR title.

**Comment by HyukjinKwon** [comment]
This fix is based upon my research wuthout any test.. so we should see the results .. :).

**Comment by HyukjinKwon** [comment]
Yes, seems much better now..

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala [hunk 1]
```diff
@@ -430,7 +430,11 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
     listFilesRecursively(new File(inputFilePath)).flatMap { file =>
       val resultFile = file.getAbsolutePath.replace(inputFilePath, goldenFilePath) + ".out"
       val absPath = file.getAbsolutePath
-      val testCaseName = absPath.stripPrefix(inputFilePath).stripPrefix(File.separator)
+      val testFilename = absPath.stripPrefix(inputFilePath).stripPrefix(File.separator)
+
+      // This is a workaround for SPARK-28894. Seems scalatest produces JUnit XML report
+      // correctly; however, Jenkins seems misunderstanding the dot in the test name.
+      val testCaseName = testFilename.stripSuffix(".sql")
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala]: Okie, so this dot (`.`) was the problem. Hm, I wonder if we have any better idea than just cutting the extension out

**Comment by HyukjinKwon** [comment]
cc @dongjoon-hyun and @maryannxue who I talked about this

**Comment by HyukjinKwon** [comment]
Let me take a look more if there's any better way.

**Comment by HyukjinKwon** [comment]
Ah .. this way will break existing testing way like :

```
build/sbt "~sql/test-only *SQLQueryTestSuite -- -z inline-table.sql"
```

Let me try to find a better way ...

**Comment by dongjoon-hyun** [comment]
Thanks for investigation. Yes. We really need to handle that properly!

**Pull request closed without merging**